app/routes/task_routes.py

```diff
@@ -1 +1,117 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request, abort, make_response, Response
+from app.models.task import Task
+from app.db import db
+from datetime import datetime
+import os
+import requests
+from app.routes.route_utilities import validate_model
+
+tasks_bp = Blueprint("task_bp", __name__, url_prefix="/tasks")
+
+@tasks_bp.post("")
+def create_task():
+    request_body = request.get_json()
+
+    if "title" not in request_body or "description" not in request_body:
+        return {"details": "Invalid data"}, 400
+
+    title = request_body["title"]
+    description = request_body["description"]
+    completed_at = request_body.get("completed_at", None)
+    is_complete = request_body.get("is_complete", False)
+    goal_id = request_body.get("goal_id", None)
+
+    new_task = Task(
+        title=title,
+        description=description,
+        is_complete=is_complete,
+        goal_id=goal_id
+    )
+    db.session.add(new_task)
+    db.session.commit()
+
+    response = {"task": new_task.to_dict()}
+    return response, 201
+
+
+@tasks_bp.get("")
+def get_all_tasks():
+    
+    sort_order = request.args.get("sort")
+
+    if sort_order == "asc":
+        query = db.select(Task).order_by(Task.title.asc())
+    elif sort_order == "desc":
+        query = db.select(Task).order_by(Task.title.desc())
+    else:
+        query = db.select(Task).order_by(Task.id)
+    
+    tasks = db.session.scalars(query)
+    tasks_response = [task.to_dict(include_goal_id=True) for task in tasks]
```

Comment:
Remember in class we wrote a function called `get_model_with_filters` though the implementation we had doesn't work in this project, the logic could be modified used here so we could DRY up our code when getting multiple records of a table.